### PR TITLE
STORM-3237 track Nimbus mkAssignment failures

### PR DESCRIPTION
**Pull request by agresch**

### CODE REVIEW

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java [hunk 1]
```diff
@@ -2850,6 +2853,7 @@ public void launchServer() throws Exception {
                                             }
                                             doCleanup();
                                         } catch (Exception e) {
+                                            this.mkAssignmentsErrors.mark();
```
Comment (kishorvpatil) [inline storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: should we have separate tracker for cleanup failures and `doCleanup()` ?

Comment (agresch) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: My thought was any exception would be worth investigating in any case.  I would hope that these issues would be rare.  Let me know if you want it more narrowly applied just to mkAssignment.

Comment (kishorvpatil) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: Yes, let's apply this narrowly to just `mkAssignments`

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java [hunk 2]
```diff
@@ -2162,7 +2165,12 @@ private boolean isReadyForMKAssignments() throws Exception {
     }
 
     private void mkAssignments() throws Exception {
-        mkAssignments(null);
+        try {
+            mkAssignments(null);
```
Comment (revans2) [inline storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: Do we want to make the change inside `mkAssignments(String scratchTopoId)` so we can count errors during the rebalance command as well?

Comment (agresch) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: I can change that.

**Comment by agresch** [comment]
@revans2 @kishorvpatil - made the change requested by Bobby.

**Pull request merged**